WolvenKit.Modkit/RED4/Tools/MeshTools.cs

```diff
@@ -661,6 +663,43 @@ public static List<RawMeshContainer> ContainRawMesh(MemoryStream gfs, MeshesInfo
             RemoveDoubleFaces(ref expMeshes);
             return expMeshes;
         }
+
+        private static string FindNextInSequence(IReadOnlyList<string> sequence, string current){
+            for (var i = 0; i < sequence.Count - 1; i++)
+            {
+                if (sequence[i] != current || sequence[i + 1] == current)
+                {
+                    continue;
+                }
+                return sequence[i + 1];
+            }
+            return null;
+        }
+
+        private static string[] InitializeDefaultMaterial(int numTotalSubmeshes, string[] meshAppearances)
+        {
+            List<string> ret = meshAppearances.ToList();;
```

Comment:
Why converting this to a list instead of doing it on the array? Below you will convert it back to array.
If you want the array's length, you can do `meshApparences.Length`.

Reply:
This avoids out of bound errors in case of arrays with a static length (which the compiler hit me with when I tried to open an ancient mesh). I figured better safe than sorry :D